diff --git a/hardware/Makefile b/hardware/Makefile
index 0625fcc..f9aa01a 100644
--- a/hardware/Makefile
+++ b/hardware/Makefile
@@ -113,8 +113,7 @@ $(buildpath):
 
 # Bender
 bender:
-	@[ -x ./bender ] && echo "Bender already exists." || \
-	curl --proto '=https' --tlsv1.2 https://fabianschuiki.github.io/bender/init -sSf | sh	-s -- 0.23.1
+	ln -sf $(BENDER) ./bender
 	@echo "$$(./bender --version) available."
 
 # Patches
